.github/workflows/ci_windows_x64_msvc.yml

```diff
@@ -22,19 +22,20 @@ concurrency:
 
 jobs:
   windows_x64_msvc:
-    runs-on: windows-2022
-    defaults:
-      run:
-        shell: bash
+    runs-on: arc-runner-set
     env:
-      BUILD_DIR: build-windows
+      BUILD_DIR_POWERSHELL: C:\mnt\azure\build-windows
     steps:
       - name: "Checking out repository"
         uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
         with:
           submodules: true
+      - name: "Clean up build dir"
+        run: |
+          if (Test-Path -Path "$Env:BUILD_DIR_POWERSHELL") {Remove-Item -Path "$Env:BUILD_DIR_POWERSHELL" -Recurse -Force}
+          mkdir "$Env:BUILD_DIR_POWERSHELL"
       - name: "Setting up Python"
-        uses: actions/setup-python@f677139bbe7f9c59b41e40162b753c062f5d49a3 # v5.1.0
+        uses: actions/setup-python@f677139bbe7f9c59b41e40162b753c062f5d49a3
```

Comment:
Keep the comment.

We'd like to use tagged versions if we _could_, but for security reasons we use hashes. Developers can freely update which commit a given tag points to, while spoofing a hash is muuuuuch harder. Picking a hash lets us know that we are using a consistent version, so if a developer pushes a malicious package, we won't immediately start using it.

The middle ground we pick is using a commit hash but putting the associated tag in a comment.